script.js

```diff
@@ -1,49 +1,64 @@
-let msgWaiting = document.querySelector(".message-waiting");
-let searchForm = document.getElementById("search-form");
-
+const msgWaiting = document.querySelector(".message-waiting");
+const searchForm = document.getElementById("search-form");
 const booksDisplay = document.getElementById("books-display");
 
-let getGoogleBooks = "https://www.googleapis.com/books/v1/volumes?q=";
-
-let apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
+const googleBooksURL = "https://www.googleapis.com/books/v1/volumes?q=";
+const apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
 
 searchForm.addEventListener("submit", () => {
   msgWaiting.style.display = "none";
-  let searchInput = document.getElementById("searchInput").value;
-  let searchResult = `${getGoogleBooks}${searchInput}${apiKey}`;
-
-  async function getBooks() {
-    const booksPromise = await fetch(searchResult);
-    const getBook = await booksPromise.json();
-    booksDisplay.innerHTML = "";
-    getBook.items.map((book) => {
-      console.log(book.volumeInfo);
-
-      let title = book.volumeInfo.title;
-      let authors = book.volumeInfo.authors;
-      let previewLink = book.volumeInfo.previewLink;
-      let publishdate = book.volumeInfo.publishedDate;
-      let publisher = book.volumeInfo.publisher;
-      let smallThumbnail = book.volumeInfo.imageLinks.smallThumbnail;
-
-      let div = document.createElement("div");
-      div.innerHTML = `
-      
-      <h3>  ${title} </h3>
-      <div class= "book-img">
-       <img src= "${smallThumbnail}" alt= "Image book cover "> 
-       </div>
-        <div class = "book-details">
+  const searchInput = document.getElementById("search-input").value;
+  const searchQuery = `${googleBooksURL}${searchInput}${apiKey}`;
+
+  getBooks(searchQuery);
+});
+
+async function getBooks(query) {
```

Comment:
It's better to move this function out of the submit event handler. That's because your event handler fires every time someone submits the form, which means your code was also re-defining `getBooks` every time the form was submitted.

Because the function is now outside the event handler, however, we need to pass it the search query.